rules/hibp/classes/rule.php

```diff
@@ -1,88 +1,116 @@
-<?php
-
-// This file is part of Moodle - http://moodle.org/
-//
-// Moodle is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-//
-// Moodle is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-//
-// You should have received a copy of the GNU General Public License
-// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
-
-/**
- * Reference implementation of a registration rule subplugin.

```

Comment:
Again maybe we can explain this is for Have I Been Pwnd, having said that I guess these are reference implementations

Reply:
That old text is not in place anymore. The whole file contained wrong line endings, so every line changed unfortunately.

Reply:
Current text was: `Registration rule restricting registrations based on data from HaveIBeenPwnd.`